### PR TITLE
feat(cli/tools/test_runner): add terse reporter

### PR DESCRIPTION
**Pull request by caspervonb**
This adds a terse reporter which displays one character per test instead of one line.

### CODE REVIEW

File: cli/tools/test_runner.rs [hunk 1]
```diff
@@ -199,8 +199,128 @@ impl TestReporter for PrettyTestReporter {
...
+          }
+        }
+      }
+
+      _ => {}
+    }
+  }
+
+  fn done(&mut self) {
+    if !self.failures.is_empty() {
```
Comment (caspervonb) [inline cli/tools/test_runner.rs:278]: Little bit of duplication here, will follow up and make these reporters stateless.

**Comment by bartlomieju** [comment]
I'm not a fan of using `--quiet` flag for "terse" output. If we were to land this PR I think we should add `--reporter` flag

**Comment by caspervonb** [comment]
> I'm not a fan of using `--quiet` flag for "terse" output. If we were to land this PR I think we should add `--reporter` flag

Mirrors `cargo test`.

Also with another flag, we'd have to call quiet anyway:

```sh
deno test --quiet --reporter=terse
```

File: cli/flags.rs [hunk 2]
```diff
@@ -1009,6 +1010,11 @@ fn test_subcommand<'a, 'b>() -> App<'a, 'b> {
         .help("Don't return error code if no test files are found")
         .takes_value(false),
     )
+    .arg(
+      Arg::with_name("terse")
+        .long("terse")
+        .help("UNSTABLE: Display one character per test instead of one line"),
```
Comment (bartlomieju) [inline cli/flags.rs:1016]: Can you add here that `--terse` implies `--quiet`?

Comment (caspervonb) [reply cli/flags.rs:1016]: Ended up not doing that after all. 
Aiming to make quiet only applies to diagnostics and terse only applies to the reporter(s).

**Review by bartlomieju — APPROVED** [review]
LGTM

**Comment by ry** [comment]
@caspervonb, first of all thanks for the PR. The feature is cleanly implemented (as always with your patches).

But I have concerns about feature creep in the test runner. Although this feature is a nice thing to have, I think one could argue it isn't a necessity. I don't like that it adds another CLI option. Is there precedence for this? The go test runner does not have this, for example.

**Comment by caspervonb** [comment]
> Is there precedence for this? The go test runner does not have this, for example.

Yeah this is pretty much nabbed directly from rust's libtest (`cargo test --quiet`).

**Pull request merged**